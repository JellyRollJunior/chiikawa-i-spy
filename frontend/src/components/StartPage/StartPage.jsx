import { Link } from 'react-router-dom';

const StartPage = () => {
  return (
    <>
      <h1>I Spy Chiikawa!</h1>
      <Link to="/game">
        <button>Start</button>
      </Link>
    </>
  );
};

export { StartPage };
