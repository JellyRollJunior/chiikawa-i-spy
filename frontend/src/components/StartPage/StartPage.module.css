.gamesTitle {
  color: var(--dark-green);
  font-size: 1.7rem;
  margin: 0 66px;
}

.gameLink {
  min-height: 200px;
  width: 300px;
}

.gameLink h3 {
  color: var(--dark-pink);
  margin-top: 3px;
}

.gameLink button {
  margin-top: 5px;
}

.gameLinkContent {
  display: block;
  padding: 8px 8px 10px 8px;
}
