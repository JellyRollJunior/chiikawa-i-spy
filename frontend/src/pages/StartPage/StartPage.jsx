import { useContext, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useGames } from '../../hooks/useGames.js';
import { NotificationContext } from '../../providers/notificationContext.jsx';
import { Header } from '../../components/Header/Header.jsx';
import { Notifications } from '../../components/Notifications/Notifications.jsx';
import { IconWrapper } from '../../components/IconWrapper/IconWrapper.jsx';
import { Leaderboard } from '../../components/Leaderboard/Leaderboard.jsx';
import { ErrorElement } from '../../components/ErrorElement/ErrorElement.jsx';
import chiikawaWoSagase from '../../assets/temp/chiikawa-wo-sagase.jpg';
import styles from './StartPage.module.css';
import shared from '../../styles/shared.module.css';
import { LoadingElement } from '../../components/LoadingElement/LoadingElement.jsx';

const StartPage = () => {
  const { games, loading, error } = useGames();
  const { addPersistentNotification } = useContext(NotificationContext);

  useEffect(() => {
    if (error) {
      addPersistentNotification(error);
    }
  }, [error]);

  return (
    <>
      <Header />
      <main>
        <Notifications />
        {loading && <LoadingElement />}
        {!loading && error && (
          <ErrorElement error={error}>
            <button
              onClick={() => window.location.reload()}
              className={`${shared.primaryButton} ${shared.marginTopSmall}`}
            >
              Refresh page
            </button>
          </ErrorElement>
        )}
        {!loading && games && !error && (
          <>
          <section>
            <div className={shared.marginTopMedium}>
              <IconWrapper size={46} margin={12}>
                <h2 className={`${shared.card} ${shared.title}`}>Games</h2>
              </IconWrapper>
            </div>
            <h3 className={`${shared.marginTopSmall} ${styles.tutorial}`}>
              Play a rousing game of I Spy with Chiikawa and friends! <br />
              Can you spot all the cute critters?
            </h3>
            <ul
              className={`${shared.horizontalWrapper} ${shared.marginTopSmall}`}
            >
              {games.map((game) => (
                <li
                  key={game.id}
                  className={`${shared.card} ${styles.gameLink}`}
                >
                  <Link
                    to={`/games/${game.id}`}
                    className={styles.gameLinkContent}
                  >
                    <img src={chiikawaWoSagase} alt="" />
                    <h3>{game.name}</h3>
                    <button className={shared.primaryButton}>
                      Start Game!
                    </button>
                  </Link>
                </li>
              ))}
            </ul>
          </section>
          </>
        )}
        <Leaderboard games={games} />
      </main>
    </>
  );
};

export { StartPage };
