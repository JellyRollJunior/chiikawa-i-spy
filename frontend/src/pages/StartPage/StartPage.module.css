.tutorial {
  color: var(--dark-green);
  text-align: center;
}

.gameLink {
  min-height: 200px;
  max-width: 450px;
}

.gameLink h3 {
  color: var(--dark-pink);
  margin-top: 3px;
}

.gameLink button {
  margin-top: 5px;
}

.gameLinkContent {
  display: block;
  padding: 8px 8px 10px 8px;
}

/* Mobile */
@media screen and (max-width: 750px) {
  .gameWrapper {
    flex-direction: column;
  }
}