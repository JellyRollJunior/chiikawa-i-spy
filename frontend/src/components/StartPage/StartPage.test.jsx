import { vi, describe, it, expect } from 'vitest';
import { screen, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { StartPage } from './StartPage.jsx';

const GAME_DATA = [
  {
    id: 1,
    name: 'Chiikawa Village',
    url: 'www.chiikawa.com',
  },
  {
    id: 3,
    name: 'Chiikawa Village - Hard',
    url: 'test',
  },
];

const mocks = vi.hoisted(() => {
  return {
    data: {},
  };
});

// Mock fetch data
vi.mock('../../hooks/useGames.js', () => ({
  useGames: () => mocks.data,
}));

// mock components
vi.mock('../Header/Header.jsx', () => ({
  Header: ({ children }) => <header>{children}</header>
}));

describe('Start page', () => {
  it('renders title and games', () => {
    // mock game data
    mocks.data = { games: GAME_DATA, loading: false, error: null };
    const { container } = render(
      <BrowserRouter>
        <StartPage />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it('renders loading text when fetching data', () => {
    // mock loading state
    mocks.data = { games: null, loading: true, error: null };
    render(
      <BrowserRouter>
        <StartPage />
      </BrowserRouter>
    );

    expect(screen.getByText('Loading games')).toBeInTheDocument();
  });

  it('renders error message when unable to fetch data', () => {
    // mock error state
    const errorMessage = 'Unable to retrieve games';
    mocks.data = { games: null, loading: true, error: errorMessage };
    render(
      <BrowserRouter>
        <StartPage />
      </BrowserRouter>
    );

    expect(screen.getByText(errorMessage)).toBeInTheDocument();
  });
});
