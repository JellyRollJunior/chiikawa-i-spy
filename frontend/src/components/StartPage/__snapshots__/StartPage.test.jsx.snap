// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Start page > renders title and games 1`] = `
<div>
  <header />
  <main>
    <section>
      <div
        class="_marginTopMedium_5890fc"
      >
        <div
          class="_iconWrapper_5d4a96"
        >
          <img
            alt="Chiikawa character icon"
            src="/src/assets/characterIcons/chiikawa-icon.png"
            style="height: 46px; width: 46px; margin-right: 12px;"
          />
          <h2
            class="_card_5890fc _title_5890fc"
          >
            Games
          </h2>
          <img
            alt="Chiikawa character icon"
            src="/src/assets/characterIcons/hachiware-icon.png"
            style="height: 46px; width: 46px; margin-left: 12px;"
          />
        </div>
      </div>
      <h3
        class="_marginTopSmall_5890fc _tutorial_eb2496"
      >
        Play a rousing game of I Spy with Chiikawa and friends! 
        <br />
        Can you spot all the cute critters?
      </h3>
      <ul
        class="_horizontalWrapper_5890fc _marginTopSmall_5890fc"
      >
        <li
          class="_card_5890fc _gameLink_eb2496"
        >
          <a
            class="_gameLinkContent_eb2496"
            data-discover="true"
            href="/games/1"
          >
            <img
              alt=""
              src="/src/assets/temp/chiikawa-wo-sagase.jpg"
            />
            <h3>
              Chiikawa Village
            </h3>
            <button
              class="_primaryButton_5890fc"
            >
              Start Game!
            </button>
          </a>
        </li>
        <li
          class="_card_5890fc _gameLink_eb2496"
        >
          <a
            class="_gameLinkContent_eb2496"
            data-discover="true"
            href="/games/3"
          >
            <img
              alt=""
              src="/src/assets/temp/chiikawa-wo-sagase.jpg"
            />
            <h3>
              Chiikawa Village - Hard
            </h3>
            <button
              class="_primaryButton_5890fc"
            >
              Start Game!
            </button>
          </a>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
